### PR TITLE
feat: ability to configure web browser user navigation headers [BISERVER-15660]

### PR DESCRIPTION
**Pull request by dcleao**
To be merged with: https://github.com/pentaho/pentaho-platform-ee/pull/2073
Issue: https://pentaho-eng.atlassian.net/browse/BISERVER-15660

### CODE REVIEW

**Review by mbrasil — APPROVED** [review]

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 39m 12s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl \
assemblies/pentaho-solutions,extensions
```
---
:ok_hand: All tests passed!
 
Tests run: **1861**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](https://wingman.eng.pentaho.com/job/projects/job/platform/job/PR-6292/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: extensions/src/main/java/org/pentaho/platform/web/servlet/matchers/UserNavigationSecFetchRequestMatcher.java [hunk 1]
```diff
@@ -115,40 +116,63 @@ public class UserNavigationSecFetchRequestMatcher implements RequestMatcher {
...
+  }
+
+  protected boolean contains( @NonNull Collection<String> values, @Nullable String value ) {
+    return value != null && values.contains( value );
   }
 
-  protected boolean contains( @NonNull List<String> list, @Nullable String value ) {
-    return value != null && list.contains( value );
+  protected List<String> toHeaderValues( @Nullable Collection<String> values, @NonNull List<String> defaultValues ) {
+    return values == null ? defaultValues : List.copyOf( values );
```
Comment (afrjorge) [inline extensions/src/main/java/org/pentaho/platform/web/servlet/matchers/UserNavigationSecFetchRequestMatcher.java]: Should it also test for empty `values` to prevent silent misconfiguration that would make the Sec-Fetch check always fail?

Comment (dcleao) [reply extensions/src/main/java/org/pentaho/platform/web/servlet/matchers/UserNavigationSecFetchRequestMatcher.java]: Made this change (with the patch you provided ;-) ).

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 42m 49s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl \
assemblies/pentaho-solutions,extensions
```
---
:ok_hand: All tests passed!
 
Tests run: **1864**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](https://wingman.eng.pentaho.com/job/projects/job/platform/job/PR-6292/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by afrjorge — APPROVED** [review]

**Comment by buildguy** [comment]
[comment]: <> (FrogbotReviewComment)

<div align='center'>

[👍 Frogbot scanned this pull request and did not find any new security issues.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 41m 57s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl \
assemblies/pentaho-solutions,extensions
```
---
:ok_hand: All tests passed!
 
Tests run: **1864**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](https://wingman.eng.pentaho.com/job/projects/job/platform/job/PR-6292/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_
